source/common/http/utility.h

```diff
@@ -39,6 +39,33 @@ class Url {
   absl::string_view path_and_query_params_;
 };
 
+class PercentEncoding {
+public:
+  /**
+   * Encodes string view to its percent encoded representation.
+   * @param value supplies string to be encoded.
+   * @return std::string percent-encoded string based on
+   * https://github.com/grpc/grpc/blob/master/doc/PROTOCOL-HTTP2.md#responses.
+   */
+  static std::string encode(absl::string_view value);
+
+  /**
+   * Encodes string view to its percent encoded representation, with start index.
+   * @param value supplies string to be encoded.
+   * @param start index supplies the first index a string should be scanned.
+   * @return std::string percent-encoded string based on
+   * https://github.com/grpc/grpc/blob/master/doc/PROTOCOL-HTTP2.md#responses.
+   */
+  static std::string encode(absl::string_view value, const size_t index);
```

Comment:
Should this be private?

Reply:
:+1:

Reply:
Done, any other things I need to change @snowp?